reactive/docker_registry.py

```diff
@@ -293,6 +293,19 @@ def update_reverseproxy_config():
         set_flag('charm.docker-registry.proxy-data.validated')
 
 
+@when('charm.docker-registry.configured')
+@when('website.available')
+@when_not('leadership.is_leader')
+def validate_follower_reverseproxy():
+    '''Remove invalid reverseproxy config for non-leader units.'''
+    # Early versions of this charm (rev 80ish) incorrectly set website config
+    # for follower units. Clean that up.
+    website = endpoint_from_flag('website.available')
+    if not is_flag_set('charm.docker-registry.proxy-data.validated'):
+        website.set_remote(all_services=None, hostname=None, port=None)
+        set_flag('charm.docker-registry.proxy-data.validated')
```

Comment:
Shouldn't you wrap this function with `@when_not('charm.docker-registry.proxy-data.validated')` ? I'm not sure I understand why the check is inside the function - which will make it call on every update-status hook as far as I can see.

Reply:
@axinojolais yup.  I originally cargo'd this from the validation done on leader units. You're right that a `@when_not` guard is better.

This also made me think that when leadership changes, we'll need to make sure we send valid website relation data.  That is, if a leader becomes a follower, we'll need to scrub the `hostname` and `port` bits, as those are only valid from a leader-perspective.

I've added a commit to address these issues.